src/App.tsx

```diff
@@ -1,9 +1,37 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+interface Event {
+  target: {
+    value: string;
+  }
+}
+
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const handleQueryChange = (event: Event) => {
```

Comment:
Let's use react types fo events.

```suggestion
  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```